Add profile name functionality to heartbeat and settings

thalesac opened the pull request:
Allow users to set a custom profile name, which can improve the identification of user sessions and preferences. E.g.
Work profile for client X, Personal, etc.

* Introduced `getDisplayProfileName` to retrieve the profile name in the heartbeat function.
* Added a new input field for custom profile name in the settings interface.
* Implemented `generateProfileIdentifier`
<!-- ELLIPSIS_HIDDEN -->

----

> [!IMPORTANT]
> Adds custom profile name functionality to heartbeat and settings, including UI input and storage handling.
> 
>   - **Behavior**:
>     - Adds `profileName` to heartbeat data in `heartbeat.ts` using `getDisplayProfileName()`.
>     - Initializes profile identifier in `main.ts` with `generateProfileIdentifier()`.
>   - **Settings UI**:
>     - Adds input field for `profileName` in `index.html`.
>     - Handles saving and restoring `profileName` in `main.ts`.
>   - **Storage**:
>     - Implements `getCustomProfileName`, `setCustomProfileName`, and `getDisplayProfileName` in `storage.ts`.
>     - Adds `generateProfileIdentifier` to create unique profile identifiers.
>   - **Misc**:
>     - Adds a custom Vite plugin in `vite.config.ts` to copy logo files during build.
> 
> <sup>This description was created by </sup>[<img alt="Ellipsis" src="https://img.shields.io/badge/Ellipsis-blue?color=175173">](https://www.ellipsis.dev?ref=ActivityWatch%2Faw-watcher-web&utm_source=github&utm_medium=referral)<sup> for 116d29b446bc854469ca00a3084385314552507c. You can [customize](https://app.ellipsis.dev/ActivityWatch/settings/summaries) this summary. It will automatically update as commits are pushed.</sup>

<!-- ELLIPSIS_HIDDEN -->

thalesac committed:
Add profile name functionality to heartbeat and settings

This change enhances user experience by allowing users to set a custom profile name, which can improve the identification of user sessions and preferences. Without this change, users would lack a personalized touch in their interactions, potentially leading to confusion when managing multiple profiles.

* Introduced `getDisplayProfileName` to retrieve the profile name in the heartbeat function.
* Added a new input field for custom profile name in the settings interface.
* Implemented `generateProfileIdentifier`

Signed-off-by: Thales Ceolin <thales@ulevel.com>

thalesac commented on vite.config.ts:
will remove that part, something happened and it didnt copy on my end. 
other than that is there anything else you want to modify? 
thanks for reviewing so quickly

BelKed replied:
I'll take a closer look in the evening, it looks fine for now 😇

thalesac replied:
excellent. just remove the unecessaru copy. 
thanks again @BelKed

BelKed replied:
Seems that there is still missing a trailing space, please remove :)

thalesac committed:
Refactor Vite configuration by removing unused logo copying logic

The removal of the custom plugin for copying the logo file streamlines the Vite configuration, reducing unnecessary complexity. Without this change, the build process could have included redundant steps that may lead to confusion or errors in future maintenance.

* Eliminated the custom plugin that copied the logo file from the media directory to the build directory.
* Removed imports for file system operations and path resolution that were no longer necessary.

Signed-off-by: Thales Ceolin <thales@ulevel.com>

BelKed reviewed (changes requested):
Hi, sorry, got a little busy in the last couple of days...

I've manually tested the change, it seems to work fine :)

I have just one request — please display the profile name in the extension popup, under **Hostname**

![](https://github.com/user-attachments/assets/72b65453-9060-4a58-be34-bedf592f10c2)

thalesac committed:
Add profile display to popup interface

This change enhances the user experience by providing visibility into the user's profile directly within the popup. Without this addition, users would lack important contextual information, potentially leading to confusion regarding their current profile status.

* Introduced a new table row in the HTML to display the user's profile.
* Integrated the `getDisplayProfileName` function to fetch the profile name from storage.
* Updated the `renderStatus` function to populate the profile name in the newly added HTML element.

Signed-off-by: Thales Ceolin <thales@ulevel.com>

thalesac commented:
@BelKed just updated with your request. THanks for looking into it!

thalesac commented:
friendly bump in case this is forgotten @BelKed

ErikBjare commented on src/background/heartbeat.ts:
```diff
@@ -28,12 +28,15 @@ async function heartbeat(
   }
 
   const now = new Date()
+  const profileName = await getDisplayProfileName()
```
Could we exclude this from data if it's something default-looking? To save on event size for 99% of users who only use the default profile.

ErikBjare commented on src/storage.ts:
```diff
@@ -101,3 +101,49 @@ export const getHostname = (): Promise<Hostname | undefined> =>
+  }
+  
+  // Generate a unique identifier for this profile/installation
+  const profileId = crypto.randomUUID()
```
No way to properly identify the profile?
